crates/services/txpool_v2/src/ports.rs

```diff
@@ -0,0 +1,79 @@
+use fuel_core_storage::Result as StorageResult;
+use fuel_core_types::{
+    entities::{
+        coins::coin::CompressedCoin,
+        relayer::message::Message,
+    },
+    fuel_tx::{
+        BlobId,
+        Bytes32,
+        ContractId,
+        UtxoId,
+    },
+    fuel_types::Nonce,
+    fuel_vm::interpreter::Memory,
+    tai64::Tai64,
+};
+
+use crate::{
+    error::Error,
+    GasPrice,
+};
+
+/// Provides an atomic view of the storage at the latest height at
+/// the moment of view instantiation. All modifications to the storage
+/// after this point will not affect the view.
+pub trait AtomicView: Send + Sync {
+    /// The type of the latest storage view.
+    type LatestView;
+
+    /// Returns current the view of the storage.
+    fn latest_view(&self) -> StorageResult<Self::LatestView>;
+}
+
+pub trait TxPoolPersistentStorage: Send + Sync {
+    fn utxo(&self, utxo_id: &UtxoId) -> StorageResult<Option<CompressedCoin>>;
+
+    fn contract_exist(&self, contract_id: &ContractId) -> StorageResult<bool>;
+
+    fn blob_exist(&self, blob_id: &BlobId) -> StorageResult<bool>;
+
+    fn message(&self, message_id: &Nonce) -> StorageResult<Option<Message>>;
+}
+
+#[async_trait::async_trait]
+/// Trait for getting gas price for the Tx Pool code to look up the gas price for a given block height
+pub trait GasPriceProvider {
+    /// Calculate gas price for the next block with a given size `block_bytes`.
```

Comment:
should `block_bytes` be a parameter to the `next_gas_price` function>

Reply:
I copy pasted from the previous pool, indeed there is no use of this in the function and so removed the doc.